evalml/tests/component_tests/test_time_series_imputer.py

```diff
@@ -562,26 +572,53 @@ def test_imputer_woodwork_custom_overrides_returned_by_components(
 
 
 @pytest.mark.parametrize(
-    "nullable_ltype",
-    ["BooleanNullable", "IntegerNullable", "AgeNullable"],
+    "nullable_ltype, expected_imputed_ltype",
+    [
+        ("BooleanNullable", Double),
+        ("IntegerNullable", Double),
+        ("AgeNullable", AgeFractional),
+    ],
 )
 def test_imputer_can_take_in_nullable_types(
     nullable_type_test_data,
     nullable_type_target,
     nullable_ltype,
+    expected_imputed_ltype,
 ):
     y = nullable_type_target(ltype=nullable_ltype, has_nans=True)
     X = nullable_type_test_data(has_nans=True)
-    # Only numeric imputing has interpolate as an option
-    X = X.ww.select("numeric")
+    # Drop the fully null columns since aren't relevant to the handle nullable types checks
+    X = X.ww.drop(["all nan", "all nan cat"])
 
-    imputer = TimeSeriesImputer(numeric_impute_strategy="interpolate")
-    imputer.fit(X, y)
-    X_imputed, y_imputed = imputer.transform(X, y)
+    cols_expected_to_change = X.ww.schema._filter_cols(
+        include=["IntegerNullable", "AgeNullable"],
+    )
+    cols_expected_to_stay_the_same = X.ww.schema._filter_cols(
+        exclude=["IntegerNullable", "AgeNullable"],
+    )
+
+    imputer = TimeSeriesImputer(
+        numeric_impute_strategy="interpolate",
+        target_impute_strategy="interpolate",
+    )
+    imputer.fit(X.ww.copy(), y)
+    X_imputed, y_imputed = imputer.transform(X.ww.copy(), y)
 
     assert not X_imputed.isnull().any().any()
     assert not y_imputed.isnull().any()
 
+    # Check that the types are as exoected
+    assert X.ww.get_subset_schema(
+        cols_expected_to_stay_the_same,
+    ) == X_imputed.ww.get_subset_schema(cols_expected_to_stay_the_same)
+    assert {
+        str(ltype)
+        for col, ltype in X_imputed.ww.logical_types.items()
+        if col in cols_expected_to_change
+    } == {"Double"}
+
+    assert isinstance(y_imputed.ww.logical_type, expected_imputed_ltype)
```

Comment:
This is interesting to me. Why do we cast `AgeNullable` to `Double` in X, but cast it to `AgeFractional` in y? Why not keep it consistent?

Reply:
I mentioned it [here](https://github.com/alteryx/evalml/pull/4046#discussion_r1128357111) - this was me not wanting to add additional complexity. 

After working on removing the nullable type logic from the rest of automl search (a separate, in progress PR), though, I'm now realizing that we'll need the logic to determine a logical type's fractional equivalent in other places as well, so I'm going to make a common util that I'll use there as well. 

I'll to pull out the [downcast_matches](https://github.com/alteryx/evalml/blob/main/evalml/utils/nullable_type_utils.py#L100) dict into a global variable `DOWNCAST_TYPE_DICT` and then to make a separate util in the nullable type utils called `determine_fractional_downcast_type` or something. So ultimately if we ever change logical types in woodwork, we only have to change that one dict.

Reply:
Also, to get at specifically why `y` was able to keep the logical type when `X` wasn't: I set `y_imputed`'s logical type to be the original downcast logical type from `_handle_nullable_types`. So if it was `AgeNullable` when passed into `transform`, if any nans were present, we'd get `AgeFractional` and then could keep that type after imputing. But if no nans were present, it'd have been downcasted to `Age`, and we can keep that type anyway since no fractional values could have been imputed. 

This kind of breaks the pattern with what we do with `X` which is to reuse the types from the original schema prior to downcasting. If that feels wrong, I should be able to do something similar with saving the original y ltype and changing it to the fractional version explicitly if need be.

Reply:
Added a common util to keep AgeFractional in X to more closely match the behavior in y, but not exactly f18ce4b1a